stories/demos/HeaderFilters.tsx

```diff
@@ -0,0 +1,191 @@
+              <option value="High">High</option>
+              <option value="Medium">Medium</option>
+              <option value="Low">Low</option>
+            </select>
+          </div>
+        )
+      },
+      {
+        key: 'issueType',
+        name: 'Issue Type',
+        filterable: true,
+        filterRenderer: p => (
+          <div className="rdg-filter-container">
+            <select className="rdg-filter" value={p.value} onChange={e => p.onChange(e.target.value)}>
+              <option value="">All</option>
+              <option value="Bug">Bug</option>
+              <option value="Improvement">Improvement</option>
+              <option value="Epic">Epic</option>
+              <option value="Story">Story</option>
+            </select>
+          </div>
+        )
+      },
+      {
+        key: 'developer',
+        name: 'Developer',
+        filterable: true,
+        width: 200,
+        filterRenderer: p => (
+          <div className="rdg-filter-container">
+            <Select
+              value={p.value}
+              onChange={p.onChange}
+              options={developerOptions}
+              styles={{
+                option: (provided) => ({
+                  ...provided,
+                  padding: 10
+                }),
+                control: (provided) => ({
+                  ...provided,
+                  height: 30,
+                  minHeight: 30,
+                  padding: 0,
+                  lineHeight: 'normal'
+                }),
+                container: (provided) => ({
+                  ...provided,
+                  width: '100%'
+                })
+              }}
+              menuPortalTarget={document.body}
+            />
+          </div>
+        )
+      },
+      {
+        key: 'complete',
+        name: '% Complete',
+        filterable: true,
+        filterRenderer: NumericFilter
+      }
+    ];
+  }, [rows]);
+
+  function filterRows(filters: Filters<Row>) {
+    return rows.filter(r => {
+      return (
+        (filters.task ? r.task.includes(filters.task) : true)
+        && (filters.priority ? r.priority === filters.priority : true)
+        && (filters.issueType ? r.issueType.includes(filters.issueType.filterTerm) : true)
+        && (filters.developer ? r.developer === filters.developer.value : true)
+        && (filters.complete ? filters.complete.filterValues(r, filters.complete, 'complete') : true)
+      );
+    });
+  }
+
+  function onToggleFilter() {
+    setEnableHeaderFilters(!enableHeaderFilters);
+    setFilters({});
+    setFilteredRows(filterRows({}));
+  }
+
+  function onFiltersChange(filters: Filters<Row>) {
+    setFilters(filters);
+    setFilteredRows(filterRows(filters));
+  }
+
+  return (
+    <>
+      <Toolbar
+        enableFilter
+        numberOfRows={rows.length}
+        onToggleFilter={onToggleFilter}
+      />
+      <AutoSizer>
+        {({ height, width }) => (
+          <DataGrid<Row, 'id'>
```

Comment:
FWIW if you type everything correctly, then TS will correctly infer the generics.